compiler/codeGen/StgCmmPrim.hs

```diff
@@ -861,6 +875,56 @@ callishPrimOpSupported dflags op
                          || llvm      -> Left MO_F64_Fabs
                      | otherwise      -> Right $ genericFabsOp W64
 
+      Pdep8Op        | (ncg && (x86ish
+                                || ppc))
+                         || llvm      -> Left (MO_Pdep    (wordWidth dflags))
+                     | otherwise      -> error "TODO: Implement (Right genericPdep8Op)"
+
+      Pdep16Op       | (ncg && (x86ish
+                                || ppc))
+                         || llvm      -> Left (MO_Pdep    (wordWidth dflags))
+                     | otherwise      -> error "TODO: Implement (Right genericPdep16Op)"
+
+      Pdep32Op       | (ncg && (x86ish
+                                || ppc))
+                         || llvm      -> Left (MO_Pdep    (wordWidth dflags))
+
+                     | otherwise      -> error "TODO: Implement (Right genericPdep32Op)"
+      Pdep64Op       | (ncg && (x86ish
+                                || ppc))
+                         || llvm      -> Left (MO_Pdep    (wordWidth dflags))
+                     | otherwise      -> error "TODO: Implement (Right genericPdep64Op)"
+
+      PdepOp         | (ncg && (x86ish
+                                || ppc))
+                         || llvm      -> Left (MO_Pdep    (wordWidth dflags))
+                     | otherwise      -> error "TODO: Implement (Right genericPdepOp)"
+
+      Pext8Op        | (ncg && (x86ish
+                                || ppc))
+                         || llvm      -> Left (MO_Pext    (wordWidth dflags))
+                     | otherwise      -> error "TODO: Implement (Right genericPext8Op)"
+
+      Pext16Op       | (ncg && (x86ish
+                                || ppc))
+                         || llvm      -> Left (MO_Pext    (wordWidth dflags))
+                     | otherwise      -> error "TODO: Implement (Right genericPext16Op)"
+
+      Pext32Op       | (ncg && (x86ish
+                                || ppc))
+                         || llvm      -> Left (MO_Pext    (wordWidth dflags))
+                     | otherwise      -> error "TODO: Implement (Right genericPext32Op)"
+
+      Pext64Op       | (ncg && (x86ish
+                                || ppc))
+                         || llvm      -> Left (MO_Pext    (wordWidth dflags))
+                     | otherwise      -> error "TODO: Implement (Right genericPext64Op)"
+
+      PextOp         | (ncg && (x86ish
+                                || ppc))
+                         || llvm      -> Left (MO_Pext    (wordWidth dflags))
+                     | otherwise      -> error "TODO: Implement (Right genericPextOp)"
+
```

Comment:
I proceeded to add the above, because it seemed like it would fix my compile errors.  It's not clear to me this is the correct approach, but put it here to ask if it makes sense.

If it is the correct thing to do, I could start working on the `error "TODO: ..."` bits, otherwise I'll revert.

Reply:
I would comment but unfortunately I can't view the patch. Have you force pushed by any chance?

Also, we generally use `panic` not `error`.

Reply:
Yes I have forced pushed.  Did that break things for you?  Should I avoid doing that?

Reply:
It seems that GitHub's pull request mechanism doesn't allow you to view old commits after a force-push. Try clicking the "View changes" button above.

Reply:
Looks like the change can only be seen from the "Files Changed" tab.